spec/services/comment_notifications_spec.rb

```diff
@@ -0,0 +1,96 @@
+describe CommentNotifications do
+  let(:user) { create :user2, :with_allies }
+  let(:ally) { user.allies.first }
+
+  context '#remove' do
+    subject { described_class.remove(params) }
+    let!(:params) do
+      {
+        comment_id: comment.id,
+        model_name: model_name,
+      }
+    end
+
+    context 'when model_name is moment' do
+      let!(:model_name) { 'moment' }
+      let!(:moment) { create(:moment, user_id: user.id, viewers: [ally.id]) }
+      let!(:comment) { Comment.create_from!(comment: 'Hello', commentable_type: model_name, commentable_id: moment.id, comment_by: ally.id, visibility: visibility) }
+
+      context 'when comment visibility is all' do
+        let!(:visibility) { 'all' }
+
+        it 'works correctly' do
```

Comment:
I'm a big fan of describing what the thing should be doing in the test because it makes it so much easier to figure out the behaviour of the code later (and for new contributors as well). I'd prefer to see something like `it 'notifies the {who or whatever its supposed to be notifying}' do`